Modules/_zoneinfo.c

```diff
@@ -2643,13 +2661,19 @@ zoneinfomodule_exec(PyObject *m)
     if (PyDateTimeAPI == NULL) {
         goto error;
     }
-    PyZoneInfo_ZoneInfoType.tp_base = PyDateTimeAPI->TZInfoType;
-    if (PyType_Ready(&PyZoneInfo_ZoneInfoType) < 0) {
+    PyObject *base = (PyObject *)PyDateTimeAPI->TZInfoType;
+    PyZoneInfo_ZoneInfoType = (PyTypeObject *)PyType_FromModuleAndSpec(m,
+                                                                       &zoneinfo_spec,
+                                                                       base);
+    if (PyZoneInfo_ZoneInfoType == NULL) {
         goto error;
     }
 
-    Py_INCREF(&PyZoneInfo_ZoneInfoType);
-    PyModule_AddObject(m, "ZoneInfo", (PyObject *)&PyZoneInfo_ZoneInfoType);
+    int rc = PyModule_AddObjectRef(m, "ZoneInfo", (PyObject *)PyZoneInfo_ZoneInfoType);
+    if (rc < 0) {
+        goto error;
+    }
+    Py_DECREF(PyZoneInfo_ZoneInfoType);
```

Comment:
refleaks in here?
If `rc` < 0 in line 2673, `Py_DECREF(PyZoneInfo_ZoneInfoType)` can not be called, no?

Reply:
True, decref should of course happen before `goto error`. Thanks!

Reply:
Fixed in 19df6f9